src/sign_tx_hash.c

```diff
@@ -6,14 +6,197 @@
+}
+
+#if defined(TARGET_STAX)
+
+static void start_review(void);
+static void ui_sign_tx_hash_nbgl(void);
+static void rejectUseCaseChoice(void);
+static nbgl_layoutTagValueList_t layout;
+static nbgl_layoutTagValue_t infos[5];  // 5 info max for ESDT and 5 info max for EGLD
+
+static const nbgl_pageInfoLongPress_t review_final_long_press = {
+    .text = "Sign transaction on\nMultiversX network?",
+    .icon = &C_icon_multiversx_logo_64x64,
+    .longPressText = "Hold to sign",
+    .longPressToken = 0,
+    .tuneId = TUNE_TAP_CASUAL,
+};
+
+static void review_final_callback(bool confirmed) {
+    if (confirmed) {
+        int tx = set_result_signature();
+        io_seproxyhal_io_heartbeat();
+        send_response(tx, true, false);
+        nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+    } else {
+        rejectUseCaseChoice();
+    }
+}
+
+static void rejectChoice(void) {
+    send_response(0, false, false);
+    nbgl_useCaseStatus("Transaction\nrejected", false, ui_idle);
+}
+
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseConfirm("Reject transaction?",
+                        NULL,
+                        "Yes, reject",
+                        "Go back to transaction",
+                        rejectChoice);
+}
+
+static void start_review(void) {
+    uint8_t step = 0;
+
+    if (should_display_esdt_flow) {
+        infos[step].item = "Token", infos[step].value = esdt_info.ticker, ++step;
+        infos[step].item = "Receiver", infos[step].value = tx_context.receiver, ++step;
+        infos[step].item = "Value", infos[step].value = tx_context.amount, ++step;
+        infos[step].item = "Fee", infos[step].value = tx_context.fee, ++step;
+        infos[step].item = "Network", infos[step].value = tx_context.network, ++step;
+    } else {
+        infos[step].item = "Receiver";
+        infos[step].value = tx_context.receiver;
+        ++step;
+        infos[step].item = "Amount";
+        infos[step].value = tx_context.amount;
+        ++step;
+        infos[step].item = "Fee";
+        infos[step].value = tx_context.fee;
+        ++step;
+        if (tx_context.data_size > 0) {
+            infos[step].item = "Data";
+            infos[step].value = tx_context.data;
+            ++step;
+        }
+        infos[step].item = "Network";
+        infos[step].value = tx_context.network;
+        ++step;
+    }
+
+    layout.nbMaxLinesForValue = 0;
+    layout.smallCaseForValue = true;
+    layout.wrapping = true;
+    layout.pairs = infos;
+    layout.nbPairs = step;
+
+    nbgl_useCaseStaticReview(&layout,
+                             &review_final_long_press,
+                             "Reject transaction",
+                             review_final_callback);
+}
+
+static void ui_sign_tx_hash_nbgl(void) {
+    if (should_display_esdt_flow) {
+        nbgl_useCaseReviewStart(&C_icon_multiversx_logo_64x64,
+                                "Review transaction to\nsend ESDT on\nMultiversX network",
```

Comment:
Also, could've used `APPNAME` here and below

Reply:
Done